app/src/main/java/org/wikipedia/activity/BaseActivity.kt

```diff
@@ -0,0 +1,292 @@
+    }
+
+    protected fun setNavigationBarColor(@ColorInt color: Int) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val isDarkThemeOrDarkBackground = (WikipediaApp.getInstance().currentTheme.isDark ||
+                    color == ContextCompat.getColor(this, android.R.color.black))
+            window.navigationBarColor = color
+            window.decorView.systemUiVisibility = if (isDarkThemeOrDarkBackground) window.decorView.systemUiVisibility and View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR.inv() else View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR or window.decorView.systemUiVisibility
+        }
+    }
+
+    protected open fun setTheme() {
+        setTheme(WikipediaApp.getInstance().currentTheme.resourceId)
+    }
+
+    protected open fun onGoOffline() {}
+    protected open fun onGoOnline() {}
+    private fun requestStoragePermission() {
+        Prefs.setAskedForPermissionOnce(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        PermissionUtil.requestWriteStorageRuntimePermissions(this@BaseActivity,
+                Constants.ACTIVITY_REQUEST_WRITE_EXTERNAL_STORAGE_PERMISSION)
+    }
+
+    private fun showStoragePermissionSnackbar() {
+        val snackbar = FeedbackUtil.makeSnackbar(this,
+                getString(R.string.offline_read_permission_rationale), FeedbackUtil.LENGTH_DEFAULT)
+        snackbar.setAction(R.string.storage_access_error_retry) { requestStoragePermission() }
+        snackbar.show()
+    }
+
+    private fun showAppSettingSnackbar() {
+        val snackbar = FeedbackUtil.makeSnackbar(this,
+                getString(R.string.offline_read_final_rationale), FeedbackUtil.LENGTH_DEFAULT)
+        snackbar.setAction(R.string.app_settings) { goToSystemAppSettings() }
+        snackbar.show()
+    }
+
+    private fun goToSystemAppSettings() {
+        val appSettings = Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS, Uri.parse("package:$packageName"))
+        appSettings.addCategory(Intent.CATEGORY_DEFAULT)
+        appSettings.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+        startActivity(appSettings)
+    }
+
+    private inner class NetworkStateReceiver : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            val isDeviceOnline = WikipediaApp.getInstance().isOnline
+            if (isDeviceOnline) {
+                if (!previousNetworkState) {
+                    onGoOnline()
+                }
+                SavedPageSyncService.enqueue()
+            } else {
+                onGoOffline()
+            }
+            previousNetworkState = isDeviceOnline
+        }
+    }
+
+    private fun removeSplashBackground() {
+        window.setBackgroundDrawable(null)
+    }
+
+    private fun unregisterExclusiveBusMethods() {
+        if (EXCLUSIVE_BUS_METHODS != null && EXCLUSIVE_DISPOSABLE != null) {
+            EXCLUSIVE_DISPOSABLE!!.dispose()
+            EXCLUSIVE_DISPOSABLE = null
+            EXCLUSIVE_BUS_METHODS = null
+        }
```

Comment:
Is it possible to use `EXCLUSIVE_DISPOSABLE?.dispose()` here and remove the `null` checks?

Reply:
correct!